### PR TITLE
[MISC] Update SeqAn3 and use sharg parser

### PR DESCRIPTION
**Pull request by Irallia**
This PR updates SeqAn3 to use the sharg parser.
As the b.i.o. lib is using an older version of SeqAn3 I had to change the dependency stuff in the CmakeLists.

Wait for https://github.com/seqan/sharg-parser/pull/111

### CODE REVIEW

**Comment by Irallia** [comment]
One test fails just for gcc10, I don't know what to do here.
```
 [ RUN      ] iGenVar_cli_test.test_genome_input
../../../iGenVar/test/cli/iGenVar_cli_test.cpp:324: Failure
Expected equality of these values:
  result.exit_code
    Which is: 11
  0
unknown file: Failure
C++ exception with description "basic_string::erase: __pos (which is 484) > this->size() (which is 0)" thrown in the test body.
[  FAILED  ] iGenVar_cli_test.test_genome_input (12 ms)
```

File: include/iGenVar.hpp [hunk 1]
```diff
@@ -1,8 +1,8 @@
 #pragma once
 
-#include <seqan3/argument_parser/argument_parser.hpp>   // for seqan3::argument_parser
+#include <sharg/parser.hpp>                     // for the SeqAn3 sharg::parser
```
Comment (smehringer) [inline include/iGenVar.hpp:3]:
```suggestion
#include <sharg/parser.hpp>                     // for the SeqAn sharg::parser
```

File: src/iGenVar.cpp [hunk 2]
```diff
@@ -13,116 +13,163 @@
...
-                      seqan3::option_spec::standard,
-                      seqan3::input_file_validator<seqan3::sequence_file_input<>>{} );
-    parser.add_option(args.output_file_path, 'o', "output",
-                      "The path of the vcf output file. If no path is given, will output to standard output.",
-                      seqan3::option_spec::standard,
-                      seqan3::output_file_validator{seqan3::output_file_open_options::open_or_create, {"vcf"}});
+    parser.add_option(args.alignment_short_reads_file_path, sharg::config{
+                        .short_id = 'i', .long_id = "input_short_reads",
+                        .description = "Input short read alignments in SAM or BAM format (Illumina).",
+                        .advanced = false,
```
Comment (smehringer) [inline src/iGenVar.cpp:64]:
```suggestion
```
you don't need this as `false` is the default. Or do you want to be explicit here?

File: src/iGenVar.cpp [hunk 3]
```diff
@@ -13,116 +13,163 @@
...
+                        .description = "The path of the optional junction output file. If no path is given, junctions will not be output.",
+                        .advanced = true,
+                        .validator = sharg::output_file_validator{sharg::output_file_open_options::open_or_create}});
+    parser.add_option(args.clusters_file_path, sharg::config{
+                        .short_id = 'b', .long_id = "clusters",
+                        .description = "The path of the optional cluster output file. If no path is given, clusters will not be output.",
+                        .advanced = true,
+                        .validator = sharg::output_file_validator{sharg::output_file_open_options::open_or_create}});
 
     // Options - Methods:
```
Comment (smehringer) [inline src/iGenVar.cpp:109]: In general, do you know of this:

```
parser.add_subsection("Methods options");
```
Could maybe add more structure to the many options?
also takes a second bolean parameter whether the section name is only displayed at the advanced help page

**Review by smehringer — APPROVED** [review]

**Pull request merged**